docs/site/Implementing-auto-migration.md

```diff
@@ -0,0 +1,158 @@
+
+```ts
+import {DataSource, Repository} from '@loopback/repository';
+import {Todo, TodoList} from './models';
+```
+
+Then get the datasource and your repositories:
+
+```ts
+const ds = await app.get<DataSource>('datasources.db');
+const todoRepo = await app.get<Repository<Todo>>('repositories.TodoRepository');
+const todoListRepo = await app.get<Repository<TodoList>>(
+  'repositories.TodoListRepository',
+);
+```
+
+Finally, call **automigrate()** or **autoupdate()**:
+
+```ts
+ds.automigrate((err: any) => {
+  if (err) throw err;
+});
+```
+
+This call to automigrate will migrate all your repositories, however if you want
+to only migrate some of your repositories, add the names of the classes in the
+first parameter:
+
+```ts
+ds.automigrate('Todo', (err: any) => {
+  if (err) throw err;
+});
+```
+
+```ts
+ds.automigrate(['Todo', 'TodoList'], (err: any) => {
+  if (err) throw err;
+});
+```
+
+**src/index.ts**
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+import {DataSource, Repository} from '@loopback/repository';
+import {Todo, TodoList} from './models';
+
+export {TodoListApplication};
+
+export async function main(options: ApplicationConfig = {}) {
+  const app = new TodoListApplication(options);
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+
+  const ds = await app.get<DataSource>('datasources.db');
+  const todoRepo = await app.get<Repository<Todo>>(
+    'repositories.TodoRepository',
+  );
+  const todoListRepo = await app.get<Repository<TodoList>>(
+    'repositories.TodoListRepository',
+  );
+
+  ds.automigrate((err: any) => {
+    if (err) throw err;
+  });
+  return app;
+}
+```
+
+### Second Approach: Individual Repositories
+
+This approach is if you want to handle migration separately in each repository.
+
+In **src/repositories/todo-list.repository.ts**, add the migrate/update call in
+the constructor:
+
+```ts
+import {Getter, inject} from '@loopback/core';
+import {
+  DefaultCrudRepository,
+  HasManyRepositoryFactory,
+  juggler,
+  repository,
+} from '@loopback/repository';
+import {Todo, TodoList} from '../models';
+import {TodoRepository} from './todo.repository';
+
+export class TodoListRepository extends DefaultCrudRepository<
+  TodoList,
+  typeof TodoList.prototype.id
+> {
+  public readonly todos: HasManyRepositoryFactory<
+    Todo,
+    typeof TodoList.prototype.id
+  >;
+
+  constructor(
+    @inject('datasources.db') dataSource: juggler.DataSource,
+    @repository.getter(TodoRepository)
+    protected todoRepositoryGetter: Getter<TodoRepository>,
+  ) {
+    super(TodoList, dataSource);
+    this.todos = this._createHasManyRepositoryFactoryFor(
+      'todos',
+      todoRepositoryGetter,
+    );
+
+    // Calling autoupdate()
+    dataSource.autoupdate('TodoList', err => {
+      if (err) throw err;
```

Comment:
Sorry, this is not going to work. When automigration fails, the error is thrown in a different tick of the event loop and will crash the application the hard way by triggering Node.js global [`uncaughtException` event](https://nodejs.org/api/process.html#process_event_uncaughtexception).

As a general rule, asynchronous operations cannot be called from a class constructor, because constructors are always sync and there is no way for the constructor to wait for the async operation to finish.

I am proposing to leave out this second approach and keep only the first one.